### PR TITLE
ZER-15 Create Calculator model

### PR DESCRIPTION
**Pull request by VictoriaBryz**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/models/calculator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Calculator < ApplicationRecord
+  has_many :fields
+  before_validation :set_uuid
+  validates :name, :presence => true
+  validates :name, :format => { :with => /\A[a-zA-Z]+\z/,
+  :message => "Only letters allowed" }
+  validates :name, :length => { :minimum => 2 }
```
Comment (maksymst) [inline app/models/calculator.rb:7]: Format your hashes per https://github.com/rubocop/ruby-style-guide#hash-literals

File: spec/models/calculator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    it { should have_many(:fields)}
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_length_of(:name).
+              is_at_least(2).
+              on(:create) }
+    it { should_not allow_value("Hh34").for(:name) }
+  end
```
Comment (maksymst) [inline spec/models/calculator_spec.rb:15]: Do not use `should` in your tests. Prefer `expect` or `is_expected`

**Review by maksymst — CHANGES_REQUESTED** [review]

File: app/models/calculator.rb (hunk 1 above)
Comment (VictoriaBryz) [reply app/models/calculator.rb]: Thanks. I've changed it in my last commit.

File: spec/models/calculator_spec.rb (hunk 2 above)
Comment (VictoriaBryz) [reply spec/models/calculator_spec.rb]: Thanks. I've changed it in my last commit.

**Review by maksymst — APPROVED** [review]
Good job 👍

**Pull request merged**